### DISCUSSION TITLE
not able to attach  tauri command to button

### DISCUSSION THREAD

**Discussion by davnav** [tauri-apps/tauri#3954] [Q&A]
I tried to attach a tauri command to  React based  "click" button, but I am getting below error. Please advise 

![image](https://user-images.githubusercontent.com/1313332/164951946-30a05a58-5188-4350-a89c-01e4b5b1b29b.png)

![image](https://user-images.githubusercontent.com/1313332/164951957-baf34435-e639-4f16-814a-b5d1e7b34aee.png)

error:

![image](https://user-images.githubusercontent.com/1313332/164952004-d09a4c45-4ace-4e61-9f7a-e5f07633e0c0.png)

![image](https://user-images.githubusercontent.com/1313332/164952019-950b3111-a4a5-427d-83b3-bbf647ea9932.png)

Do I need to make any changes in  tauri.conf.json  ?

After saving the rust code,  I am just running yarn start  . Please advise if I missed any step ?
Thank you

**Comment by nothingismagick** [comment]
Please always check that you are on the latest version by running `tauri info` with yarn, npm, pnpm, or cargo. Also, please share your tauri.conf.json

**Reply by davnav** [reply]
![image](https://user-images.githubusercontent.com/1313332/164984999-ce8450c0-cb46-4021-9290-67b32db9742a.png)





```
{
  "package": {
    "productName": "tauriexp",
    "version": "0.1.0"
  },
  "build": {
    "distDir": "../build",
    "devPath": "http://localhost:3000",
    "beforeDevCommand": "yarn start",
    "beforeBuildCommand": "yarn build"
  },
  "tauri": {
    "bundle": {
      "active": true,
      "targets": "all",
      "identifier": "com.tauri.dev",
      "icon": [
        "icons/32x32.png",
        "icons/128x128.png",
        "icons/128x128@2x.png",
        "icons/icon.icns",
        "icons/icon.ico"
      ],
      "resources": [],
      "externalBin": [],
      "copyright": "",
      "category": "DeveloperTool",
      "shortDescription": "",
      "longDescription": "",
      "deb": {
        "depends": [],
        "useBootstrapper": false
      },
      "macOS": {
        "frameworks": [],
        "useBootstrapper": false,
        "exceptionDomain": "",
        "signingIdentity": null,
        "providerShortName": null,
        "entitlements": null
      },
      "windows": {
        "certificateThumbprint": null,
        "digestAlgorithm": "sha256",
        "timestampUrl": ""
      }
    },
    "updater": {
      "active": false
    },
    "allowlist": {
      "all": true
    },
    "windows": [
      {
        "title": "Mytauri",
        "width": 800,
        "height": 600,
        "resizable": true,
        "fullscreen": false
      }
    ],
    "security": {
      "csp": null
    }
  }
}
```

Above is  my tauri.conf.json

**Comment by FabianLars** [comment]
tauri's IPC only works in tauri windows, not in your browser. To start tauri in dev mode you need to run `yarn tauri dev` instead of `yarn start` (if `yarn start` is not set as a beforeDevCommand in tauri.conf.json then you need to run that in a different terminal too).

If you absolutely need to have it working in a browser (at dev time) a combination of <https://github.com/tauri-apps/tauri-plugin-localhost> and <https://github.com/tauri-apps/tauri-invoke-http> should work (but no promises that it works with the latest rc versions)

**Reply by davnav** [reply]
Below is the logs for yarn tauri dev.
looks like build failed. is that mean my rust program has some issue !

![image](https://user-images.githubusercontent.com/1313332/164984647-2fa08fde-071b-45fb-b34a-9e0fec747c56.png)
